File: androidApp/src/main/java/bisq/client/android/presentation/notification_list/components/NotificationListItem.kt
package bisq.client.android.presentation.notification_list.components

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material.MaterialTheme
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import bisq.client.android.presentation.components.icons.*
import bisq.client.domain.model.Notification
import bisq.client.domain.model.NotificationType
import bisq.client.domain.util.DatetimeUtil
import kotlinx.coroutines.ExperimentalCoroutinesApi

@ExperimentalCoroutinesApi
@ExperimentalStdlibApi
@Composable
fun NotificationListItem(
    notification: Notification,
    onClick: () -> Unit
) {
    val datetimeUtil = remember { DatetimeUtil() }
    Column(
        modifier = Modifier
            .padding(
                bottom = 6.dp,
                top = 6.dp,
            )
            .fillMaxWidth()
            .clickable(onClick = onClick),
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(top = 8.dp, bottom = 0.dp, start = 8.dp, end = 8.dp)
        ) {
            when (NotificationType.valueOf(notification.type)) {
                is NotificationType.PeerMessage -> MessageIcon()
                is NotificationType.MarketOffer -> BitcoinIcon()
                is NotificationType.OfferTaken -> OfferIcon()
                is NotificationType.PriceAlert -> PriceIcon()
                is NotificationType.TradeState -> TradeIcon()
                else -> UnknownIcon()
            }
            Text(
                text = notification.message,
                style = MaterialTheme.typography.h5,
                modifier = Modifier
                    .padding(start = 8.dp)
            )
        }
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(top = 0.dp, bottom = 8.dp, start = 38.dp, end = 8.dp)
        ) {
            Text(
                text = datetimeUtil.prettyDatetime(notification.timeOccurred),
                style = MaterialTheme.typography.caption
            )
        }
    }
}

File: iosApp/iosApp/presentation/notification_list/components/NotificationListItem.swift
import SwiftUI
import common

struct NotificationListItem: View {
    
    private let notification: common.Notification
    private let dateUtil: DatetimeUtil
    
    init(
        notification: common.Notification,
        dateUtil: DatetimeUtil
    ) {
        self.notification = notification
        self.dateUtil = dateUtil
    }
    
    var body: some View {
        HStack() {
            switch NotificationType.companion.valueOf(name: notification.type) {
                case is NotificationType.PeerMessage:
                    MessageIcon()
                case is NotificationType.MarketOffer:
                    BitcoinIcon()
                case is NotificationType.OfferTaken:
                    OfferIcon()
                case is NotificationType.PriceAlert:
                    PriceIcon()
                case is NotificationType.TradeState:
                    TradeIcon()
                default:
                    UnknownIcon()
            }
            VStack(alignment: .leading) {
                //DefaultText(String(notification.message))
                DefaultText(String(NotificationType.companion.toString(type: NotificationType.companion.valueOf(name: notification.type))))
                DefaultText("\(dateUtil.prettyDatetime(date: notification.timeOccurred))")
                    .foregroundColor(Color.gray)
            }
        }
    }
}
